cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1376,20 +1386,39 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
          */
         Future<ReconciliationState> zkScaleUpInSmallSteps() {
             return zkSetOperations.getAsync(namespace, ZookeeperCluster.zookeeperClusterName(name))
-                    .map(sts -> sts == null ? Integer.valueOf(0) : sts.getSpec().getReplicas())
-                    .compose(currentReplicas -> {
-                        if (currentReplicas > 0 && zkCluster.getReplicas() > currentReplicas) {
-                            zkCluster.setReplicas(currentReplicas + 1);
-                            zkScalingUp = true;
-                        }
+                    .compose(sts -> {
+                        if (sts != null) {
+                            int currentReplicas = sts.getSpec().getReplicas();
+                            if (currentReplicas > 0 && zkCluster.getReplicas() > currentReplicas) {
+                                // TODO: For ZK 3.5.x we need to limit scaling to 1 at a time. Should we enforce that here?
+                                zkCluster.setReplicas(currentReplicas + 1);
+                                zkScalingUp = true;
+                            } else if (currentReplicas > 0 && zkCluster.getReplicas() < currentReplicas) {
+                                zkScalingDown = true;
+                            }
 
+                            if ((zkScalingUp || zkScalingDown) && (KafkaVersion.compareDottedVersions(zkCluster.getVersion(), "3.4.99") > 0)) {
+                                log.info(
+                                        "Scaling Zookeeper 3.5.x cluster: Rolling updates will be skipped and the manual " +
+                                                "update process defined in the documentation should be followed"
+                                );
+                                // Setting this will add the annotation to the STS in the state.zkStatefulSet() method
+                                zkManualScaling = true;
+                            } else if (Annotations.booleanAnnotation(sts, Annotations.ANNO_STRIMZI_IO_MANUAL_ZK_SCALE, false)) {
```

Comment:
What if the anno is missing? Will this trigger or what will it do?

Reply:
That branch will only be hit if we are not scaling or if we are scaling a 3.4.14 cluster. In which case if the anno is not there it will return false and the ZK roll will proceed as `zkManualScaling` will be false. I will add an else clause at the end to make sure it is false.